### ISSUE TITLE
`CreateObject` menu: rename `Engine` path to `oM.NonBHoMObjects`

### ISSUE THREAD

**Issue by epignatelli** [BHoM/BHoM_UI#92]
The menu on the `CreateObject` component now shows two main levels: `oM` and `Engine, to separate out methods that output `BHoMObject`s and methods that don't.

Renaming the `Engine` branch into `oM.Non-BHoM-Object` will make ti a child branch of oM (and then prune it altogether).

**Comment by epignatelli** [comment]
To have a feel:
![image](https://user-images.githubusercontent.com/26899347/58093117-ba8bab00-7bc5-11e9-8aea-44b20f7c6a5a.png)
 
I prefer `SoftwareSpecific` as it is less BHoM-centric.
@al-fisher @rwemay @adecler @IsakNaslundBh @alelom

**Comment by alelom** [comment]
> I prefer `SoftwareSpecific` as it is less BHoM-centric.

I agree, especially considering user-friendliness. Drawback is that this could be problematic when you have something that is neither software-specific nor BHoM-related. However, I'd go for `SoftwareSpecific` anyway.

**Comment by IsakNaslundBh** [comment]
> I prefer SoftwareSpecific as it is less BHoM-centric.

Not sure I do. I mean, we have reflection methods in there. They are not really software specific. `NonBHoMObject` maybe? And probably with the `Object` work on the end. Everything in there is still part of the BHoM ecosystem after all.

**Comment by alelom** [comment]
> Not sure I do. I mean, we have reflection methods in there. They are not really software specific. `NonBHoMObject` maybe? And probably with the `Object` word on the end. Everything in there is still part of the BHoM ecosystem after all.

My point in favour of `SoftwareSpecific` is because it feels 1) Less BHoM-centric 2) More intuitive, user-friendly. However, I know it's not a good definition. This is also why, among the other options mentioned during our last meeting, somebody proposed `Advanced`. Another option we mentioned was `Other`.

Thing is, I think we've already discussed these things during our meeting, and I remember we eventually concurred on `NonBHoM`, so I think @IsakNaslundBh has it right.

I just wanted to make clear that I get @epignatelli's point. Maybe `Other` would be a more agreeable solution?
